js/spreadsheet.js

```diff
@@ -68,244 +68,196 @@ var spreadsheet = function(options) {
   }
 
   /**
-   * We user this method to determine where is closest data is lokated from the selected cell
+   * Function to check is there any data in the nearest cells
    * 
-   * @param {Array} selectedCell - array of the coordinat of the selected cell reacived throw the hot.getSelected() method
+   * @param {object} selectedCell - object with keys row and col which contains selected cell coordinates
+   * @param {array} tableData - an array of the table data from Handsontable library
    * 
-   * @returns {Object} - object of the directions where data is placed occording to the selected cell
+   * @returns {object} - possible keys: right, left, top, bottom - shows which closest cells have data
    */
-  function closestData(selectedCell) {
-    selectedCell = selectedCell[0];
-    if (!Array.isArray(selectedCell)) {
-      console.error('We must pass an array of the cell coordinats to the closestData function. First element is cell' +
-        'row and second element is cell col. In this case script will act as if there was a value in the cell. ' +
-        'Value that was passed - ',
-        selectedCell);
-      return false;
-    }
-
-    var col = selectedCell[1];
-    var row = selectedCell[0];
-    var selectedCellData = hot.getDataAtCell(row, col)
-    // At this block we getting an index of the nearest cells from the selected cell
-    // If we selected first row it ID is 0 already and if we - 1 from it we will reacive an error in the hot.getDataAtCell() method
-    var top = row ? row - 1 : row;
-    var bottom = row + 1;
-    // Same as in top variable
-    var left = col ? col - 1 : col;
-    var right = col + 1;
-    // At this block we reacive the nearest cells value from the selected cell
-    var leftValue = hot.getDataAtCell(row, left);
-    var rightValue = hot.getDataAtCell(row, right);
-    var topValue = hot.getDataAtCell(top, col);
-    var bottomValue = hot.getDataAtCell(bottom, col);
-    var dataAt = {
-      left: false,
-      right: false,
-      top: false,
-      bottom: false,
-      all: false,
-      hasData: false
+  function nearestCellsWithData(selectedCell, tableData) {
+    var cellsWithData = {
+      right: tableData[selectedCell.row][selectedCell.col + 1] !== null,
+      left: tableData[selectedCell.row][selectedCell.col - 1] !== null,
+      top: tableData[selectedCell.row - 1][selectedCell.col] !== null,
+      bottom: tableData[selectedCell.row + 1][selectedCell.col] !== null,
```

Comment:
Add a `current: tableData[selectedCell.row][selectedCell.col] !== null`, then just return the `cellsWithData` object. In fact, it doesn't really a `var` declared.

The reason for this is that the function will do exactly what the name suggests, and does not add logic on top to dictate what other functions should do with the information. The caller of this function can then decide under what scenario it should _select all_ etc.